modules/overrides/userconfigurable/client/duration.go

```diff
@@ -39,7 +41,13 @@ func (d *Duration) UnmarshalJSON(input []byte) error {
 	return nil
 }
 
+func (d *Duration) Fake(faker *gofakeit.Faker) (any, error) {
```

Comment:
Are fakes used for production functionality or only tests?  If only tests, then would prefer if there is a way to keep this dependency isolated to only the tests.